src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -478,16 +476,19 @@ public String getName() {
   @Override
   public void getDocIds(DocIdPusher pusher) throws IOException,
       InterruptedException {
+    DateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS Z");
+    formatter.setTimeZone(updateTimestampTimezone.getTimeZone());
     BufferedPusher outstream = new BufferedPusher(pusher);
+    lastModifiedCache.invalidateAll();
```

Comment:
I think I would move this to the first line (it's tied to the semantics of getDocIds, not this try statement or the loop, and these fields/variables are used in this order below: lastModifiedCache, formatter, outstream).